src/js/controller/controller.js

```diff
@@ -336,13 +338,37 @@ Object.assign(Controller.prototype, {
         function _checkPlayOnViewable(model, viewable) {
             if (model.get('playOnViewable')) {
                 if (viewable) {
-                    _autoStart();
-                } else if (OS.mobile || (model.get('autoPause') && model.get('autoPause').viewability)) {
+                    _autoStart('viewable');
+                } else if (OS.mobile) {
                     _this.pause({ reason: 'autostart' });
                 }
             }
         }
 
+        function _checkPauseOnViewable(model, viewable) {
+            const instream = _this._instreamAdapter;
+
+            // If player left view and came back before ad completes,
+            // the player should begin/resume playing content.
+            if (viewable && instream && instream.noResume) {
+                instream.noResume = false;
+                model.set('playOnViewable', true);
+                return;
+            } else if (viewable || !model.get('autoPause').viewability) {
+                return;
+            }
+
+            const adState = _getAdState();
+            if (!adState) {
+                _this.pause({ reason: 'viewable' });
+            } else if (model.get('activeTab') && adState !== 'paused') {
+                if (instream) {
+                    instream.noResume = true;
+                }
+                model.set('playOnViewable', true);
+            }
+        }
```

Comment:
There's too much repetition and disconnected logic in this function. You should rewrite it.

Here's what a simplified version looks like: 
```js
if (model.get('autoPause').viewability) {
    _model.change('viewable', _checkPauseOnViewable);
}
...

function _checkPauseOnViewable(model, viewable) {
    const adState = _getAdState();
    if (adState) {
         _this._instreamAdapter. noResume = !viewable;
    }
    model.set('playOnViewable', !viewable);
    if (!viewable) {
        _this.pause({ reason: 'viewable' });
    }
}
```

Reply:
Can't do this because it will pause if an ad is currently playing.

Reply:
The code above may not be complete. But it is a much smaller template for you to base the logic on. Complete the logic as necessary, without splitting the logic for instream an ad state in multiple places.